tactics/btermdn.ml

```diff
@@ -130,9 +131,8 @@ let constr_pat_discr env ts t =
   | PVar v, args ->
     begin match ts with
     | None -> None
```

Comment:
Why is `PVar` different from `PRef VarRef` when ts=None?
Why does ts=None give false in evaluable_named but true in evaluable_constant?
Are these questions related to the new FIXME?

Reply:
The `PRef VarRef` case is essentially dead code, there might be a way to trigger it with handwritten patterns in hints but even there it's not clear. For the rest, see #14753.